Source/HtmlTextWriter/HtmlTextWriter.cs

```diff
@@ -249,6 +283,22 @@ public void WriteAttribute(string name, string value, bool encode)
                 this.Write($"{EqualsDoubleQuoteString}{value}{DoubleQuoteChar}");
             }
         }
+        
+        //************************************************
+        public void WriteStyleAttribute(string name, string value) => this.WriteStyleAttribute(name, value, true);
+        public void WriteStyleAttribute(string name, string value, bool encode)
+        {
+            this.Write($"{name}{StyleEqualsChar}");
+
+            if (value != null)
+            {
+                if (encode)
+                    value = WebUtility.HtmlEncode(value);
+
+                this.Write($"{value}{SemicolonChar} ");
```

Comment:
If a style attribute's value is null, shouldn't the semicolon be written for it anyway?